tests/sles4sap/trento/test_hana_stop.pm

```diff
@@ -11,35 +11,62 @@ use base 'consoletest';
 use testapi;
 use serial_terminal 'select_serial_terminal';
 use utils 'script_retry';
-use qesapdeployment qw(qesap_upload_logs qesap_ansible_cmd);
+use qesapdeployment;
 use trento;
 
 
 sub run {
     my ($self) = @_;
     select_serial_terminal;
 
+    my $primary_host = 'vmhana01';
+
     my $prov = get_required_var('PUBLIC_CLOUD_PROVIDER');
-    qesap_ansible_cmd(cmd => 'crm status', provider => $prov, filter => 'vmhana01');
-    qesap_ansible_cmd(cmd => 'SAPHanaSR-showAttr', provider => $prov, filter => 'vmhana01');
+    qesap_ansible_cmd(
+        cmd => 'crm status',
+        provider => $prov,
+        filter => $primary_host);
+    qesap_ansible_cmd(cmd => 'SAPHanaSR-showAttr --format=script',
+        provider => $prov,
+        filter => $primary_host);
+
+    # Stop the primary DB
+    qesap_ansible_cmd(
+        cmd => "su - hdbadm -c 'HDB stop'",
+        provider => $prov,
+        filter => $primary_host);
 
-    qesap_ansible_cmd(cmd => "su - hdbadm -c 'HDB stop'", provider => $prov, filter => 'vmhana02');
+    my $_monitor_start_time = time();
+    my $done;
+    while ((time() - $_monitor_start_time <= 300) && (!$done)) {
+        sleep 10;
+        my $show_attr = qesap_ansible_script_output(
+            cmd => 'SAPHanaSR-showAttr',
+            provider => $prov,
+            host => $primary_host,
+            root => 1);
 
-    my $cmr_status = qesap_ansible_script_output(cmd => 'crm status', provider => $prov, host => 'vmhana01', root => 1);
-    record_info("crm status", $cmr_status);
-    qesap_ansible_cmd(cmd => 'SAPHanaSR-showAttr', provider => $prov, filter => 'vmhana01');
+        my @status = ();
+        for my $line (split("\n", $show_attr)) {
+            $status[0] = $line if (($line =~ m/^vmhana01/));
+            $status[1] = $line if (($line =~ m/^vmhana02/));
```

Comment:
Suggestion:
```
my %status = ();
for my $line (split("\n", $show_attr)) {
    $status{$1} = $line if ($line =~ m/^(vmhana\d{2})/);
}
```
And re-write L54 & L55 accordingly.

This way you can use `$status{vmahana01}` instead of `$status[0]` which IMO is more readable.